code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -921,108 +990,94 @@ pub mod pallet {
 		#[transactional]
 		fn split(
 			who: &Self::AccountId,
-			(fnft_collection_id, fnft_instance_id): &Self::PositionId,
+			(fnft_collection_id, old_fnft_instance_id): &Self::PositionId,
+			// REVIEW(benluelo): Rename this argument so it's clearer what the split is doing.
+			// Perhaps something like `keep_ratio`?
 			ratio: Permill,
-		) -> Result<[Self::PositionId; 2], DispatchError> {
-			// TODO(benluelo): Refactor
-			let keep_alive = false;
-			let mut old_position_stake = BalanceOf::<T>::zero();
-			let mut old_position =
-				Stakes::<T>::try_mutate(fnft_collection_id, fnft_instance_id, |maybe_stake| {
-					let stake = maybe_stake.as_mut().ok_or(Error::<T>::StakeNotFound)?;
-
-					let old_value = stake.clone();
-					stake.stake = ratio.mul_floor(stake.stake);
-					stake.share = ratio.mul_floor(stake.share);
+		) -> Result<Self::PositionId, DispatchError> {
+			let (new_fnft_instance_id, new_position) = Stakes::<T>::try_mutate(
+				fnft_collection_id,
+				old_fnft_instance_id,
+				|maybe_old_position| {
+					let old_position =
+						maybe_old_position.as_mut().ok_or(Error::<T>::StakeNotFound)?;
+
+					let left_from_one_ratio = ratio.left_from_one();
+
+					let new_stake = left_from_one_ratio.mul_ceil(old_position.stake);
+					let new_share = left_from_one_ratio.mul_ceil(old_position.share);
+					let new_reductions = {
+						let mut r = old_position.reductions.clone();
+						for (_, reduction) in &mut r {
+							*reduction = left_from_one_ratio.mul_floor(*reduction);
+						}
+						r
+					};
 
-					for (_, reduction) in &mut stake.reductions {
+					old_position.stake = ratio.mul_floor(old_position.stake);
+					old_position.share = ratio.mul_floor(old_position.share);
+					for (_, reduction) in &mut old_position.reductions {
 						*reduction = ratio.mul_floor(*reduction);
 					}
 
-					old_position_stake = stake.stake;
-					Ok::<_, DispatchError>(old_value)
-				})?;
-
-			let left_from_one_ratio = ratio.left_from_one();
-
-			for (_, reduction) in &mut old_position.reductions {
-				*reduction = left_from_one_ratio.mul_floor(*reduction);
-			}
-
-			let new_fnft_instance_id = T::FinancialNft::get_next_nft_id(fnft_collection_id)?;
-			let old_fnft_asset_account =
-				T::FinancialNft::asset_account(fnft_collection_id, &old_position.fnft_instance_id);
-			let new_fnft_asset_account =
-				T::FinancialNft::asset_account(fnft_collection_id, &new_fnft_instance_id);
-
-			let new_stake = StakeOf::<T> {
-				stake: left_from_one_ratio.mul_floor(old_position.stake),
-				share: left_from_one_ratio.mul_floor(old_position.share),
-				fnft_instance_id: new_fnft_instance_id,
-				..old_position
-			};
+					let rewards_pool = RewardPools::<T>::get(old_position.reward_pool_id)
+						.ok_or(Error::<T>::RewardsPoolNotFound)?;
+
+					// REVIEW(benluelo): Make this one operation? (Would require a change in the
+					// fNFT interface)
```

Comment:
Perhaps we should create an ergonomic interface ontop of fNFT for use in other pallets while leaving the base fNFT pallet as is.

Reply:
I think that would definitely be the best bet, @vimukthi-git and i briefly talked about this recently. The main issue is i'm not sure what the interface should be (pallet, trait, etc)

Reply:
We want to keep `mint_into` as a standard interface as used by Substrate so let's keep this as is :)